draft-ietf-quic-transport.md

```diff
@@ -1483,10 +1483,13 @@ subsequent sections.
 ## STREAM Frame {#frame-stream}
 
 STREAM frames implicitly create a stream and carry stream data. The type byte
-for a STREAM frame contains embedded flags, and is formatted as `11FSSOOD`.
+for a STREAM frame contains embedded flags, and is formatted as `1UFSSOOD`.
 These bits are parsed as follows:
 
-* The first two bits must be set to 11, indicating that this is a STREAM frame.
+* The first bit must be set to 1, indicating that this is a STREAM frame.
+
+* 'U' is the UNI bit, which is used to indicate this is a unidirectional
+  stream, and no stream frames may be sent in response.
```

Comment:
No, I think you've invented a new problem.

Consider how IMAP works now: the client pulls up a TCP connection to the server and then the server's first message is the advertisement.

Now consider a hypothetical new version of IMAP which runs over QUIC but allows (for some reason) multiple concurrent IMAP sessions from the client. In this model, the client knows how many sessions it wants (the server has no idea) and it opens up a new stream for each of them. In the current QUIC model, the server can start sending the banner on each one of those connections immediately as soon as it knows it exists, whether it has received a STREAM frame or not. However, in your new model, if there is any way to have a unidirectional stream, then the server can't know whether a given stream is bidirectional and represents a new IMAP session or unidirectional. Now, you can say that the IMAP binding forbids unidirectional streams, but that doesn't stop them from being tagged that way in the STREAM frame and now you have an anomaly.